enterprise/server/githubapp/githubapp.go

```diff
@@ -80,21 +82,133 @@ func Register(env *real_environment.RealEnv) error {
 	if !*enabled {
 		return nil
 	}
-	app, err := New(env)
+	readWriteApp, err := NewReadWriteApp(env)
 	if err != nil {
 		return err
 	}
-	env.SetGitHubApp(app)
+	a, err := NewAppService(env, readWriteApp)
+	if err != nil {
+		return err
+	}
+	env.SetGitHubAppService(a)
 	return nil
 }
 
 func IsEnabled() bool {
 	return *enabled
 }
 
-// GitHubApp implements the BuildBuddy GitHub app. Users install the app to
-// their personal account or organization, granting access to some or all
-// repositories.
+// GitHubAppService is a wrapper for GitHubApp. Because there are 2 BuildBuddy
+// GitHub apps (read-only vs read-write), it helps determine the specific app
+// the user has installed.
+type GitHubAppService struct {
+	env environment.Env
+
+	readWriteApp interfaces.GitHubApp
+	// TODO(Maggie): Add read only app
+}
+
+func NewAppService(env environment.Env, readWriteApp interfaces.GitHubApp) (*GitHubAppService, error) {
+	return &GitHubAppService{
+		env:          env,
+		readWriteApp: readWriteApp,
+	}, nil
+}
+
+// GetGitHubApp returns the BB GitHub app that the current user has authorized.
+func (s *GitHubAppService) GetGitHubApp(ctx context.Context) (interfaces.GitHubApp, error) {
+	u, err := s.env.GetUserDB().GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if u.GithubToken == "" {
+		return nil, status.NotFoundErrorf("no linked GitHub account was found")
+	}
+	// If the user has already linked an app installation, check its app ID.
+	// Check our database first over using the GitHub API, because the API is
+	// rate limited.
+	installations, _ := s.GetGitHubAppInstallations(ctx)
```

Comment:
should we log.Error here if there's an error? it shouldn't normally fail, even if there are 0 installations linked